src/pkg/cli/composeUp.go

```diff
@@ -96,10 +107,22 @@ func ComposeUp(ctx context.Context, loader client.Loader, c client.FabricClient,
 			}
 			deployRequest.DelegationSetId = delegation.DelegationSetId
 		}
+
+		timestamp := time.Now()
 		resp, err = p.Deploy(ctx, deployRequest)
 		if err != nil {
 			return nil, project, err
 		}
+		err = c.PutDeployment(ctx, &defangv1.Deployment{
```

Comment:
I guess the main concern here would be the Deploy() occurring then the CLI exiting for some reason and we don't have a chance to run PutDeployment(). Is there any major downsides if we happen to miss a deployment history entry?

Reply:
It's a problem either way: we write the deployment to Fabric but Deploy fails (validation?) or we deploy and are unable to write to Fabric. For BYOC, writing deployments is just a log, with no real meaning, so I think the small race where deploy start but CLI aborts before we "Put" is acceptable.

Reply:
I would argue it probably makes more sense to make the `PutDeployment` call before the `Deploy` call, so we always have a record on the deployment attempts.
Or even more complicated, but not necessary, make 2 `PutDeployment` calls before and after?

Reply:
I'd like to keep this as a simple record of deployment requests that we have sent. I don't see any benefit to making the same call again.

No matter the order, I think it is very unlikely that the client can send one request successfully and not send the other. They are both very quick to respond. (In fact, we can send them concurrently!)